React/Modules/RCTAsyncLocalStorage.m

```diff
@@ -445,4 +392,31 @@ - (NSDictionary *)_writeEntry:(NSArray<NSString *> *)entry changedManifest:(BOOL
   }
 }
 
+RCT_EXPORT_METHOD(setStorageDirectory:(NSString*)storageDirectory
+                  callback:(RCTResponseSenderBlock)callback)
+{
+  storageDirectory = [storageDirectory stringByAppendingPathComponent:RCTStorageDirectory];
+  if(cachedStorageDirectory){
+    if([cachedStorageDirectory compare:storageDirectory] == NSOrderedSame){
+      return;
+    }
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSError *error = nil;
+    if([fileManager fileExistsAtPath:storageDirectory]){
```

Comment:
We probably need to distinguish between switching storage location and migrating to a new storage location, as these are separate concepts.